src/sqlfluff/rules/L013.py

```diff
@@ -47,42 +48,67 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
+            return None
+
+        # If this isnt a select_clause then this rule doesnt apply
+        if not segment.all(sp.is_type("select_clause_element")):
+            return None
+
+        # Ignore if it's a function with EMITS clause as EMITS is equivalent to AS
+        if (
+            children.select(sp.is_type("function"))
+            .children()
+            .select(sp.is_type("emits_segment"))
         ):
-            # Ignore if it's a function with EMITS clause as EMITS is equivalent to AS
-            if (
-                children.select(sp.is_type("function"))
-                .children()
-                .select(sp.is_type("emits_segment"))
-            ):
-                return None
-
-            types = set(
-                children.select(sp.not_(sp.is_name("star"))).apply(sp.get_type())
-            )
-            unallowed_types = types - {
-                "whitespace",
-                "newline",
-                "column_reference",
-                "wildcard_expression",
-            }
-            if unallowed_types:
-                # No fixes, because we don't know what the alias should be,
-                # the user should document it themselves.
-                if self.allow_scalar:  # type: ignore
-                    # Check *how many* elements there are in the select
-                    # statement. If this is the only one, then we won't
-                    # report an error.
-                    immediate_parent = context.functional.parent_stack.last()
-                    num_elements = len(
-                        immediate_parent.children(sp.is_type("select_clause_element"))
-                    )
-                    if num_elements > 1:
-                        return LintResult(anchor=context.segment)
-                    else:
-                        return None
-                else:
-                    # Just error if we don't care.
-                    return LintResult(anchor=context.segment)
-        return None
+            return None
+
+        select_clause_children = children.select(sp.not_(sp.is_name("star")))
+        is_complex_clause = _recursively_check_is_complex(select_clause_children)
+        if not is_complex_clause:
+            return None
+        # No fixes, because we don't know what the alias should be,
+        # the user should document it themselves.
+        if self.allow_scalar:  # type: ignore
+            # Check *how many* elements/columns there are in the select
+            # statement. If this is the only one, then we won't
+            # report an error.
+            immediate_parent = context.functional.parent_stack.last()
+            elements = immediate_parent.children(sp.is_type("select_clause_element"))
+            num_elements = len(elements)
+
+            if num_elements > 1:
+                return LintResult(anchor=context.segment)
+            return None
+
+        # Just error if we don't care.
```

Comment:
Should this comment be removed or changed? Previously, the comment was present when the `return` was inside `if self.allow_scalar`. I'm not sure it makes sense anymore. (Did it ever?)